### DISCUSSION TITLE
API key isn't working for me. What am I doing wrongly?

### DISCUSSION THREAD

**Discussion by bec-aus** [amberelectric/public-api#162] [Q&A] [answered]
Hi,
I'm trying to use the API key from Home Assistant and I get a message saying "Invalid API key".  I intially tried with HA on a Synology NAS, now have tried it on a Raspberry Pi. Same response either way.
I tried with 2 different API tokens. (& I set them up a few days ago, so systems have had time to synch.)

I went hunting to try to troubleshoot it and noticed that the For Developers page in Amber has a way to try to test using the API.  I entered the key in there and I get a 403 error with a message "User is not authorized to access this resource with an explicit deny".  from this request:
`
curl -X 'GET' \
  'https://api.amber.com.au/v1/sites' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer a1b2358ff136795237fdbf4c95e3a1a1'
`
NB I have changed characters in the token a for a letter and 1 for a number, but it is the same length as the API token I took from the website.

Also I tested this
`
curl -X 'GET' \
  'https://api.amber.com.au/v1/state/vic/renewables/current?next=1&previous=1&resolution=30' \
  -H 'accept: application/json'
` 
and it returns sensible information

So I assume that I have grabbed the wrong code to use as a key. Is there any guidance as to what is correct?

**Comment by madpilot** [comment, marked as answer]
The API key starts with psk_ - it’s possible you didn’t cut and paste it correctly

**Reply by bec-aus** [reply]
Thanks, I had what I thought was the key right. I didn’t expect that prefix to be part of it. Once the key is saved it disappears so I couldn’t even sit and stare at it wondering what I had missed. 

Perhaps for dopes like me the ‘format’ of the key could be shown on the page. Or a simple note like yours. Eg the API token starts with psk_ and includes the following xx (16?) characters.
